tests/framework/installer/ceph_installer.go

```diff
@@ -795,3 +810,52 @@ spec:
                   path: /run/udev
 `
 }
+
+func (h *CephInstaller) addCleanupPolicy(namespace string) error {
+	cluster, err := h.k8shelper.RookClientset.CephV1().CephClusters(namespace).Get(namespace, metav1.GetOptions{})
+	if err != nil {
+		return fmt.Errorf("failed to get ceph cluster. %+v", err)
+	}
+	cluster.Spec.CleanupPolicy.DeleteDataDirOnHosts = "yes-really-destroy-data"
+	_, err = h.k8shelper.RookClientset.CephV1().CephClusters(namespace).Update(cluster)
+	if err != nil {
+		return fmt.Errorf("failed to add clean up policy to the cluster. %+v", err)
+	}
+	logger.Info("successfully added cleanup policy to the ceph cluster")
+	return nil
+}
+
+func (h *CephInstaller) waitForCleanupJobs(namespace string) error {
+	allRookCephCleanupJobs := func() (done bool, err error) {
+		appLabelSelector := fmt.Sprintf("app=%s", cluster.CleanupAppName)
+		cleanupJobs, err := h.k8shelper.Clientset.BatchV1().Jobs(namespace).List(metav1.ListOptions{LabelSelector: appLabelSelector})
+		if err != nil {
+			return false, fmt.Errorf("failed to get cleanup jobs. %+v", err)
+		}
+		// Clean up jobs might take some time to start
+		if len(cleanupJobs.Items) == 0 {
+			logger.Infof("no jobs with label selector %q found.", appLabelSelector)
+			return false, nil
+		}
+		for _, job := range cleanupJobs.Items {
+			logger.Debugf("job %q status: %+v", job.Name, job.Status)
+			if job.Status.Failed > 0 {
+				return false, fmt.Errorf("job %s failed", job.Name)
+			}
+			if job.Status.Succeeded == 0 {
+				return false, nil
+			}
+		}
+		logger.Infof("cleanup job(s) completed")
+		return true, nil
+	}
+
+	logger.Info("waiting for job(s) to cleanup the host...")
+	err := wait.Poll(5*time.Second, 90*time.Second, allRookCephCleanupJobs)
+	if err != nil {
+		return fmt.Errorf("failed to wait for clean up jobs to complete. %+v", err)
+	}
+
+	logger.Info("successfully executed all the ceph clean up jobs")
```

Comment:
This test verifies that the cleanup jobs run, but it doesn't verify that the cleanup jobs actually clean up directories. It is also probably a good idea to verify that they do *not* clean up directories that shouldn't be cleaned up like might belong to a different ceph cluster.

Reply:
When we are using this exclusively for cleanup between integration test suites I think we will naturally get this check as well. I'd suggest we hold off on changing the integration test validation from this PR.